### PR TITLE
refactored kamel.ts and fixed some errors caused by incorrect catches

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/kamel.ts [hunk 1]
```diff
@@ -14,15 +14,14 @@
...
-import { exec, spawn} from "child_process";
+import { exec, spawn, ChildProcess} from "child_process";
 import * as child_process from "child_process";
 import * as config from './config';
 import * as utils from './CamelKJSONUtils';
 import * as extension from './extension';
 import * as path from 'path';
 import * as fs from 'fs';
-import * as shell from './shell';
+import { findBinary } from './kubectl';
```
Comment (apupier) [inline src/kamel.ts:24]: That's good to remove duplicated code.
I think the `findBinary` should be extracted to its own class. There is no reason that kamel has a dependency on kubectl.

**Comment by lhein** [comment]
> can you precise which "errors are fixed" that were caused by "incorrect catches"?

see https://travis-ci.org/github/camel-tooling/vscode-camelk/builds/709054714 
lines 645-718

File: src/kamel.ts (hunk 1 above)
Comment (lhein) [reply src/kamel.ts]: agreed. moved it to the shell.ts

**Comment by lhein** [comment]
waiting for @bfitzpat  's review before merging

**Pull request merged**